### ISSUE TITLE
Tf unconnected trees

### ISSUE THREAD

**Issue by horverno** [rosblox/ros_gz_sim_nav2_outdoor_example#3]
hi @maxpolzin

I tried to make the simulation up and running the standard way (clone, colcon build, rosdep install, source). Both rviz2 and  gazebo starts, but i got the following error:

``` r
[controller_server-11] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: 
Could not find a connection between 'odom' and 'base_link' because they are not part of the same tree. 
Tf has two or more unconnected trees.
```

What did I miss? 

thanks

**Comment by maxpolzin** [comment]
Did you wait for some time? The robot_localization nodes, i.e. navsat_transform and ukf, sometimes need some time to settle.

They both work together to provide this transform, odom to base_link.

**Comment by maxpolzin** [comment]
Hi @horverno. Thx for trying. 

Did you wait for some time? The robot_localization nodes, i.e. navsat_transform and ukf, sometimes need some time to settle.

They both work together to provide this transform, odom to base_link.


Another thing that comes to my mind: which version of ignition gazebo do you have installed, particularly the sensors package?

**Comment by horverno** [comment]
thanks for the quick reply, yes i waited a several minutes, also checked with `rqt_tf_tree`, which confirms the unconnected tree:

![image](https://github.com/rosblox/nav2_outdoor_example/assets/11504709/b49e0e07-bb29-4660-b345-b65aa80562e4)

**Comment by maxpolzin** [comment]
Can you check your versions of the installed ignition gazebo (sensors) packages?

**Comment by horverno** [comment]
i don't know if that's the correct way but i checked with:
```
dpkg -l | grep ignition-sensors
```
results:
``` r
ii  libignition-sensors6:amd64                         6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Shared library
ii  libignition-sensors6-air-pressure:amd64            6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib air-pressure
ii  libignition-sensors6-air-pressure-dev:amd64        6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files air-pressure
ii  libignition-sensors6-altimeter:amd64               6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib altimeter
ii  libignition-sensors6-altimeter-dev:amd64           6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files altimeter
ii  libignition-sensors6-boundingbox-camera:amd64      6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib Bounding Box Camera
ii  libignition-sensors6-boundingbox-camera-dev:amd64  6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files Bounding Box Camera
ii  libignition-sensors6-camera:amd64                  6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib Camera
ii  libignition-sensors6-camera-dev:amd64              6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files Camera
ii  libignition-sensors6-core-dev:amd64                6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Development files
ii  libignition-sensors6-depth-camera:amd64            6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib depth-camera
ii  libignition-sensors6-depth-camera-dev:amd64        6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files depth-camera
ii  libignition-sensors6-dev:amd64                     6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Metapackage
ii  libignition-sensors6-force-torque:amd64            6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib force-torque
ii  libignition-sensors6-force-torque-dev:amd64        6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files force-torque
ii  libignition-sensors6-gpu-lidar:amd64               6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib gpu-lidar
ii  libignition-sensors6-gpu-lidar-dev:amd64           6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files gpu-lidar
ii  libignition-sensors6-imu:amd64                     6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib imu
ii  libignition-sensors6-imu-dev:amd64                 6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files imu
ii  libignition-sensors6-lidar:amd64                   6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib lidar
ii  libignition-sensors6-lidar-dev:amd64               6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files lidar
ii  libignition-sensors6-logical-camera:amd64          6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib logical-camera
ii  libignition-sensors6-logical-camera-dev:amd64      6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files logical-camera
ii  libignition-sensors6-magnetometer:amd64            6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib magnetometer
ii  libignition-sensors6-magnetometer-dev:amd64        6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files magnetometer
ii  libignition-sensors6-navsat:amd64                  6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib navsat
ii  libignition-sensors6-navsat-dev:amd64              6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files navsat
ii  libignition-sensors6-rendering:amd64               6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib rendering
ii  libignition-sensors6-rendering-dev:amd64           6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files rendering
ii  libignition-sensors6-rgbd-camera:amd64             6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib rgbd-camera
ii  libignition-sensors6-rgbd-camera-dev:amd64         6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files rgbd-camera
ii  libignition-sensors6-segmentation-camera:amd64     6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib segmentation-camera
ii  libignition-sensors6-segmentation-camera-dev:amd64 6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files segmentation-camera
ii  libignition-sensors6-thermal-camera:amd64          6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Lib thermal-camera
ii  libignition-sensors6-thermal-camera-dev:amd64      6.7.0-1~jammy                           amd64        Gazebo Sensors classes and functions for robot apps - Dev files thermal-camera
```

**Comment by maxpolzin** [comment]
These are my versions. 

I think I have installed Gazebo Fortress from the Gazebo repo, https://gazebosim.org/docs/latest/ros_installation and https://gazebosim.org/docs/fortress/install_ubuntu 

I think this installs the newer versions with updated and fixes for the sensor plugins.

![image](https://github.com/rosblox/nav2_outdoor_example/assets/20051567/c32e53f3-a723-43aa-b0c4-5b064748127e)

**Comment by horverno** [comment]
Thanks! That was it. I removed gazebo with `sudo apt remove ignition-fortress && sudo apt autoremove` and re-intalled, now it works!

Also a sidenote, I still have older gazebo sensors, but the main point is that it works now. 
``` r
sudo apt list --installed | grep imu
libignition-sensors6-imu-dev/unknown,now 6.7.1-1~jammy amd64 [installed,automatic]
libignition-sensors6-imu/unknown,now 6.7.1-1~jammy amd64 [installed,automatic]
```

**Comment by maxpolzin** [comment]
Great.

I think it was fixed with the changes from 6.7.0 to 6.7.1.
